manager/state/raft/transport/peer.go

```diff
@@ -29,25 +29,49 @@ type peer struct {
 	cancel context.CancelFunc
 	done   chan struct{}
 
-	mu      sync.Mutex
-	cc      *grpc.ClientConn
-	addr    string
-	newAddr string
+	mu    sync.Mutex
+	cc    *grpc.ClientConn
+	addr  string
+	addrs map[string]struct{}
 
 	active       bool
 	becameActive time.Time
 }
 
-func newPeer(id uint64, addr string, tr *Transport) (*peer, error) {
-	cc, err := tr.dial(addr)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create conn for %x with addr %s", id, addr)
+func newPeer(id uint64, tr *Transport, addrs ...string) (*peer, error) {
+	if len(addrs) == 0 {
+		return nil, errors.New("addrs list can't be empty")
+	}
+	var (
+		cc   *grpc.ClientConn
+		addr string
+	)
+	for _, a := range addrs {
+		c, err := tr.dial(a)
+		if err != nil {
+			log.G(tr.ctx).WithError(err).Errorf("failed to create conn for %x with addr %s", id, a)
+			continue
+		}
+		addr = a
+		cc = c
+		break
```

Comment:
I slightly prefer to structure the loop so it doesn't end with `break` (`if err == nil { ... }`) but it's not a big deal.